source/packaging/ext_list.cpp

```diff
@@ -0,0 +1,140 @@
+#include <xlnt/packaging/ext_list.hpp>
+#include <algorithm>
+
+#include <detail/external/include_libstudxml.hpp>
+
+namespace {
+    // send elements straight from parser to serialiser without modification
+    // runs until the end of the current open element
+    xlnt::uri roundtrip(xml::parser& p, xml::serializer &s)
+    {
+        xlnt::uri ext_uri;
+        int nest_level = 0;
+        while (nest_level > 0 || (p.peek() != xml::parser::event_type::end_element && p.peek() != xml::parser::event_type::eof))
+        {
+            switch (p.next())
+            {
+            case xml::parser::start_element:
+            {
+                ++nest_level;
+                auto attribs = p.attribute_map();
+                s.start_element(p.qname());
+                if (nest_level == 1)
+                {
+                    ext_uri = xlnt::uri(attribs.at(xml::qname("uri")).value);
+                }
+                auto current_ns = p.namespace_();
+                p.peek(); // to look into the new namespace
+                auto new_ns = p.namespace_(); // only before attributes?
+                if (new_ns != current_ns)
+                {
+                    auto pref = p.prefix();
+                    s.namespace_decl(new_ns, pref);
+                }
+                for (auto &ele : attribs)
+                {
+                    s.attribute(ele.first.string(), ele.second.value);
+                }
+                break;
+            }
+            case xml::parser::end_element:
+            {
+                --nest_level;
+                s.end_element();
+                break;
+            }
+            case xml::parser::start_namespace_decl:
+            {
+                s.namespace_decl(p.namespace_(), p.prefix());
+                break;
+            }
+            case xml::parser::end_namespace_decl:
+            { // nothing required here
+                break;
+            }
+            case xml::parser::characters:
+            {
+                s.characters(p.value());
+                break;
+            }
+            case xml::parser::eof:
+                return ext_uri;
+            case xml::parser::start_attribute:
+            case xml::parser::end_attribute:
+            default:
+                break;
+            }
+        }
+        return ext_uri;
+    }
+}
+
+namespace xlnt {
+
+ext_list::ext::ext(xml::parser &parser, const std::string& ns)
+{
+    std::ostringstream serialisation_stream;
+    xml::serializer s(serialisation_stream, "", 0);
+    s.start_element(xml::qname(ns, "wrap")); // wrapper for the xmlns declaration
+    s.namespace_decl(ns, "");
+    extension_ID_ = roundtrip(parser, s);
+    s.end_element(xml::qname(ns, "wrap"));
+    serialised_value_ = serialisation_stream.str();
```

Comment:
While this works, it's not a particularly clean solution. Is there a better way to store the <ext...> elements?

Reply:
This is a tricky one. The other two options I've considered are: only support round-tripping for a subset of extLst types and store the children in a tree data structure or string writing it back directly during serialization. In any case, this seems fine for now. I'll add an issue for improving it.